sed/loader/flash/loader.py

```diff
@@ -786,11 +731,29 @@ def parquet_handler(
                 detector,
             )
 
-            # Read all parquet files using dask and concatenate into one dataframe after filling
-            dataframe = self.fill_na(
-                [dd.read_parquet(file) for file in parquet_filenames],
+            # Read all parquet files into one dataframe using dask
+            dataframe = dd.read_parquet(parquet_filenames)
+
+            # Channels to fill NaN values
+            channels: List[str] = self.get_channels_by_format(["per_pulse", "per_train"])
+
+            # Define a custom function to forward fill specified columns
+            def forward_fill_partition(df):
+                df[channels] = df[channels].ffill()
+                return df
+
+            # calculate the number of rows in each partition and choose least
+            nrows = dataframe.map_partitions(len)
+            max_part_size = min(nrows)
```

Comment:
This is here because we cannot prepend data from two partitions away. Therefore the only single number which always works is the minimum size of all partitions.
Maybe there is a way to make map_overlap use always the size of the previous/next partition, but I am afraid not.